experimental/Schemes/SpaceGerms.jl

```diff
@@ -152,59 +425,411 @@ function SpaceGerm(X::AbsSpec, I::MPolyIdeal)
+        over rational field
+      by ideal(x1^2 + x2^2 + x3^2, x1*x2)
+    at complement of maximal ideal of point (0, 0, 0)
+
+```
+"""
+function CompleteIntersectionGerm(X::AbsSpec, a::Vector{T}) where T<:Union{Integer, FieldElem}
+  R = ambient_coordinate_ring(X)
+  kk = coefficient_ring(R)
+  b = [kk.(v) for v in a]  ## throws an error, if vector entries are not compatible
+  U = MPolyComplementOfKPointIdeal(R,b)
+  L,_ = Localization(OO(X), U)
+  mingens = minimal_generating_set(modulus(L))
+## TODO: Should be dim(L) and not dim(Spec(L)), but dim for localized 
+## quotients is only repaired on the geometric side as of now!!!
+  SpecL = Spec(L)
+  length(mingens) == dim(R) - dim(SpecL) || error("not a complete intersection")
+  w = mingens
+  Y = CompleteIntersectionGerm(SpecL,w)
+  set_attribute!(Y,:representative,X)
+  return Y
+end
+
+function CompleteIntersectionGerm(X::AbsSpec, I::MPolyIdeal)
+  R = base_ring(I)
+  R === ambient_coordinate_ring(X) || error("rings are not compatible")
+  a = rational_point_coordinates(I)
+  Y = CompleteIntersectionGerm(X,a)
+  return Y
+end
+
+function ComleteIntersectionGerm(X::AbsSpec, I::Ideal)
+  A = base_ring(I)
+  A === OO(X) || error("rings are incompatible")
+  J = saturated_ideal(I)
+  a = rational_point_coordinates(J)
+  return CompleteIntersectionGerm(X,a)
 end
 
-function germ_at_point(A::MPolyRing, I::Ideal)
-  X = Spec(A)
+@doc raw"""
+    CompleteIntersectionGerm(X::AbsSpec, p::AbsAffineRationalPoint)
+    CompleteIntersectionGerm(p::AbsAffineRationalPoint)
+
+Returns a complete intersection germ `(X,p)` for a given `X`and a rational point `p` on some affine scheme `Y`, provided that $X$ is locally a complete intersection in some neighbourhood of `p`. If no `X` is specified, `Y` is used in its place.
+"""
+CompleteIntersectionGerm(p::AbsAffineRationalPoint) = CompleteIntersectionGerm(codomain(p), coordinates(p))
+
+function CompleteIntersectionGerm(X::AbsSpec, p::AbsAffineRationalPoint)
+  ambient_space(X) == ambient_space(codomain(p)) || error("ambient spaces do not match")
+  return CompleteIntersectionGerm(X,coordinates(p))
+end
+
+@doc raw"""
+    germ_at_point(X::AbsSpec, I::Union{Ideal,Vector})
+    germ_at_point(X::AbsSpec, I::Union{Ideal,Vector})
+    germ_at_point(A::Union{MPolyRing,MPolyQuoRing},
+              I::Union{Ideal,Vector})
+    germ_at_point(A::LocalRing, I::Union{Ideal,Vector})
+
+Returns a SpaceGerm `(X,p)` and the corresponding inclusion morphism of spectra
+arising from the given representative `X` or the given
+`X = Spec(A)` for a local ring `A`, where the point `p` may be specified in several
+equivalent ways:
+- by its coordinates `a` in the ambient_space of `X` or
+- by a maximal ideal `I`in the coordinate ring of `X` or
+- by a maximal ideal `I` in the ambient_coordinate_ring of `X`
+- by the maximal ideal of the local ring `A`
+
+!!!note: Only `LocalRing`s localized at rational points over the coefficient field are currently supported.
```

Comment:
Why?

Reply:
I added the work fully before supported -- we do not have standard bases in this case, so we cannot really calculate. Hence there are strong restriction on the prime ideal case.